### PR TITLE
Feat/login page

### PR DESCRIPTION
**Pull request by alekhya66**

### CODE REVIEW

File: src/Components/fe-login/FeLogin.js [hunk 1]
```diff
@@ -37,14 +48,36 @@ export class FeLogin extends LocalizeMixin(LitElement) {
...
+    try {
+      const { serializedValue } = ev.target;
+      await FeServices.postRequest({
+        url: '/login',
+        data: {
+          login: serializedValue.login,
+        },
+      });
+      this.dispatchEvent(
+        new CustomEvent('input-validation', { detail: serializedValue })
```
Comment (Rajkumar-MP) [inline src/Components/fe-login/FeLogin.js]: why its need for dispatch ? and why u pass the login details, what is the use ?

Comment (Rajkumar-MP) [reply src/Components/fe-login/FeLogin.js]: where is the response data ?, we need the data based on that only account call will trigger

File: src/Components/fe-login/test/fe-login.test.js [hunk 2]
```diff
@@ -21,6 +32,27 @@ describe('FeLogin', () => {
...
+    const requestMock = sinon.stub(ajax, 'requestJson');
+    requestMock.rejects({});
+    username.modelValue = 'username';
+    password.modelValue = 'password';
+
+    element.login();
+    await aTimeout(10);
+    requestMock.restore();
+    expect(notification.classList.contains('hidden')).to.be.false;
+    expect(username.modelValue).to.equal('username');
```
Comment (Rajkumar-MP) [inline src/Components/fe-login/test/fe-login.test.js]: what is the use of this expect ?, you only added the modelValue and you itself checking that same as equal ?

**Review by Rajkumar-MP — CHANGES_REQUESTED** [review]

File: src/Components/fe-login/test/fe-login.test.js (hunk 2 above)
Comment (alekhya66) [reply src/Components/fe-login/test/fe-login.test.js]: ok will change

File: src/Components/fe-login/FeLogin.js (hunk 1 above)
Comment (Nayana-2607) [reply src/Components/fe-login/FeLogin.js]: Made the changes and uploaded

File: src/Components/fe-login/stories/index.stories.js [hunk 3]
```diff
@@ -7,9 +7,7 @@ export default {
 };
 
 function Template() {
-  return html`
-    <fe-login @input-validation=${ev => console.log(ev.detail)}> </fe-login>
-  `;
+  return html` <fe-login> </fe-login> `;
```
Comment (Rajkumar-MP) [inline src/Components/fe-login/stories/index.stories.js]: You need to check the login details event

File: src/Pages/LoginPage/LoginPage.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    try {
+      const data = await FeServices.postRequest({
+        url: '/login',
+        data: {
+          username: detail.username,
+          password: detail.password,
+        },
+      });
+
+      this.dispatchEvent(new CustomEvent('id', { detail: data.id }));
```
Comment (Rajkumar-MP) [inline src/Pages/LoginPage/LoginPage.js]: Instead of generic id, use some specific name like customer-id or other something like that

**Review by Rajkumar-MP — CHANGES_REQUESTED** [review]

File: src/Components/fe-login/stories/index.stories.js (hunk 3 above)
Comment (alekhya66) [reply src/Components/fe-login/stories/index.stories.js]: okay will check

File: src/Pages/LoginPage/LoginPage.js (hunk 4 above)
Comment (alekhya66) [reply src/Pages/LoginPage/LoginPage.js]: okay the id will be changed to customer-id

File: src/Components/fe-login/stories/index.stories.js (hunk 3 above)
Comment (Rajkumar-MP) [reply src/Components/fe-login/stories/index.stories.js]: once completed, then make it resolve

Comment (Nayana-2607) [reply src/Components/fe-login/stories/index.stories.js]: resolved and pushed the
code

**Review by Rajkumar-MP — DISMISSED** [review]

**Review by Rajkumar-MP — APPROVED** [review]

**Review by Nayana-2607 — DISMISSED** [review]

**Review by shadushazzz — DISMISSED** [review]

File: src/Pages/LoginPage/LoginPage.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import { html, css, LitElement } from 'lit-element';
+import defaultStyles from '../../FeApp.style.js';
+import { FeServices } from '../../FeServices.js';
+
```
Comment (Rajkumar-MP) [inline src/Pages/LoginPage/LoginPage.js:6]: you didnt import the login component here, it will fail with application level

**Review by Rajkumar-MP — CHANGES_REQUESTED** [review]

File: src/Pages/LoginPage/LoginPage.js (hunk 5 above)
Comment (Rajkumar-MP) [reply src/Pages/LoginPage/LoginPage.js:6]: and notification import also

Comment (Nayana-2607) [reply src/Pages/LoginPage/LoginPage.js:6]: I will import them and update

File: src/Components/fe-login/FeLogin.js [hunk 6]
```diff
@@ -20,6 +21,11 @@ export class FeLogin extends LocalizeMixin(LitElement) {
     `;
   }
 
+  constructor() {
```
Comment (Rajkumar-MP) [inline src/Components/fe-login/FeLogin.js]: constructor is not needed, bcz we are not doing anything from `isError`

**Review by Rajkumar-MP — CHANGES_REQUESTED** [review]

File: src/Components/fe-login/FeLogin.js (hunk 6 above)
Comment (Nayana-2607) [reply src/Components/fe-login/FeLogin.js]: okay will remove the constructor

**Review by Rajkumar-MP — DISMISSED** [review]

**Review by Nayana-2607 — DISMISSED** [review]

**Review by shadushazzz — DISMISSED** [review]

**Comment by alekhya66** [comment]
feat login page

**Review by Nayana-2607 — APPROVED** [review]

**Review by shadushazzz — APPROVED** [review]

**Review by shadushazzz — APPROVED** [review]

**Review by Rajkumar-MP — APPROVED** [review]

**Pull request merged**